torch/csrc/jit/passes/pass_manager.h

```diff
@@ -23,17 +23,77 @@ namespace jit {
 // A pass modifies a Graph in place.
 using GraphPass = std::function<void(std::shared_ptr<Graph>&)>;
 
-TORCH_API std::vector<GraphPass>& getCustomPostFusionPasses();
-TORCH_API std::vector<GraphPass>& getCustomPreFusionPasses();
+// Since Passes are std::functions, we associate a UUID to each pass, this way
+// if we want to deregister a pass, we have something to reference it by.
+using GraphPassNameType = unsigned int;
+// Start UUID at 1
+static GraphPassNameType graphPassID = 1;
 
-struct TORCH_API RegisterPostFusionPass {
-  RegisterPostFusionPass(GraphPass p);
-};
+// Graph pass entries have a name associated with them
+using GraphPassEntry = std::pair<GraphPass, GraphPassNameType>;
+
+// Return currently registered passes. Passes are stored in a static vector
+TORCH_API std::vector<std::pair<GraphPass, GraphPassNameType>>& getCustomPostPasses();
+TORCH_API std::vector<std::pair<GraphPass, GraphPassNameType>>& getCustomPrePasses();
+
+TORCH_API GraphPassNameType registerPostPass(GraphPass p);
+TORCH_API GraphPassNameType registerPrePass(GraphPass p);
+//Default to registerPostPass
+TORCH_API GraphPassNameType registerPass(GraphPass p);
```

Comment:
I kept `registerPass` when added `registerPostPass`/`registerPrePass` to not break it for users who might have been using the old API. This PR changes its signature anyway, so we can remove `registerPass` altogether, no reason to keep it.

Reply:
Removed.